src/core/device.c

```diff
@@ -155,12 +155,19 @@ static int device_coldplug(Unit *u) {
 
         assert(d);
         assert(d->state == DEVICE_DEAD);
+        assert(d->found == DEVICE_NOT_FOUND);
 
-        /* First, let's put the deserialized state and found mask into effect, if we have it. */
+        /* Note that during the "coldplug" phase (called after enumeration and
+         * deserialization ones) no unit state change events are generated, thus the
+         * deserialized state won't pull in new deps */
 
-        if (d->deserialized_state < 0 ||
-            (d->deserialized_state == d->state &&
-             d->deserialized_found == d->found))
+        /* Device appeared while PID1 stopped listening for new kernel events, which
+         * happens while it is reexecuting */
+        if (d->deserialized_state == _DEVICE_STATE_INVALID)
```

Comment:
this isn't really a state. that's why it has an underscore... it's supposed to mean "invalid", and I think we really should stick to checking for invalid with a "< 0" check here, to cover anything negative really, because invalid just means invalid, and shouldn't be a specific value.

Reply:
Ok